src/app/admin/createCollectionCentre/CreateCollectionCentreForm.tsx

```diff
@@ -47,21 +46,24 @@ const CreateCollectionCentreForm: React.FC<{}> = () => {
         setSubmitDisabled(true);
 
         if (checkErrorOnSubmit(formErrors, setFormErrors)) {
-            setSubmitError(Errors.submit);
             setSubmitDisabled(false);
             return;
         }
 
         const { error } = await supabase.from("collection_centres").insert(fields);
 
         if (error) {
-            setSubmitError(Errors.external);
-            setSubmitErrorMessage(error.message);
+            if (error.code === "23505") {
+                setSubmitErrorMessage(
+                    "A Collection Centre with this name/abbreviation has already been added. Please choose a different name/abbreviation"
+                );
```

Comment:
Can we instead have a function that takes in the error code and returns an optional error message outside the component, and call it from here to decide whether to show a custom message?
```
function ...(errorCodeOnCreateAttempt: string): string | null {
} 
```

Reply:
Think I have now done what you asked